src/sources/redis/list.rs

```diff
@@ -0,0 +1,111 @@
+                                    });
+                                    let event = create_event(line.as_str(),key.clone(),&redis_key);
+                                    tokio::select!{
+                                        result = out.send(event) => {match result {
+                                            Ok(()) => { },
+                                            Err(()) => return Ok(()),
+                                        }}
+                                        _ = &mut shutdown => return Ok(()),
+                                    }
+                                }
+                                Err(error) => {
+                                    error!(message = "Redis source generated an error.", %error);
+                                    emit!(&RedisReceiveEventFailed { error });
+                                }
+                            }
+                        }
+                        _ = &mut shutdown => return Ok(()),
+                    }
+                }
+            };
+            Box::pin(fut)
+        }
+        Method::Blpop => {
+            let fut = async move {
+                trace!("Get redis connection manager.");
+                let mut conn = client
+                    .get_tokio_connection_manager()
+                    .await
+                    .expect("Failed to get redis async connection.");
+                trace!("Get redis connection manager success.");
+                loop {
+                    tokio::select! {
+                        res = blpop(&mut conn,key.as_str()) => {
+                            match res {
+                                Ok(line) => {
+                                    emit!(&RedisEventReceived {
+                                        byte_size: line.len()
+                                    });
+                                    let event = create_event(line.as_str(),key.clone(),&redis_key);
+                                    tokio::select!{
+                                        result = out.send(event) => {match result {
+                                            Ok(()) => { },
+                                            Err(()) => return Ok(()),
+                                        }}
+                                        _ = &mut shutdown => return Ok(()),
+                                    }
+                                }
+                                Err(error) => {
+                                    emit!(&RedisReceiveEventFailed {error});
+                                }
+                            }
+                        }
+                        _ = &mut shutdown => return Ok(()),
+                    }
+                }
+            };
+            Box::pin(fut)
+        }
+    }
+}
+
+async fn brpop(conn: &mut ConnectionManager, key: &str) -> RedisResult<String> {
+    let res: RedisResult<(String, String)> = conn.brpop(key, 0).await;
```

Comment:
I think it would be nice to make the `timeout` argument configurable from the source config.

Reply:
The timeout argument  specifying the maximum number of seconds to block. A timeout of zero can be used to block indefinitely. I think this parameter does not need to be configured

Reply:
@pablosichert 
The timeout argument is specifying the maximum number of seconds to block. A timeout of zero can be used to block indefinitely. I think this parameter does not need to be configured